contracts/cw20-streams/src/contract.rs

```diff
@@ -0,0 +1,575 @@
+use crate::error::ContractError;
+use crate::msg::{
+    ConfigResponse, ExecuteMsg, InstantiateMsg, ListStreamsResponse, QueryMsg, ReceiveMsg,
+    StreamParams, StreamResponse,
+};
+use crate::state::{save_stream, Config, Stream, CONFIG, STREAMS, STREAM_SEQ};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Order, Response, StdResult,
+    Uint128,
+};
+use cw2::set_contract_version;
+use cw20::{Cw20Contract, Cw20ExecuteMsg, Cw20ReceiveMsg};
+use cw_storage_plus::Bound;
+
+const CONTRACT_NAME: &str = "crates.io:cw20-streams";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let owner = match msg.owner {
+        Some(own) => deps.api.addr_validate(&own)?,
+        None => info.sender,
+    };
+
+    let config = Config {
+        owner: owner.clone(),
+        cw20_addr: deps.api.addr_validate(&msg.cw20_addr)?,
+    };
+    CONFIG.save(deps.storage, &config)?;
+
+    STREAM_SEQ.save(deps.storage, &0u64)?;
+
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("owner", owner)
+        .add_attribute("cw20_addr", msg.cw20_addr))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Receive(msg) => execute_receive(env, deps, info, msg),
+        ExecuteMsg::Withdraw { id } => execute_withdraw(env, deps, info, id),
+    }
+}
+
+pub fn execute_create_stream(
+    env: Env,
+    deps: DepsMut,
+    config: Config,
+    params: StreamParams,
+) -> Result<Response, ContractError> {
+    let StreamParams {
+        owner,
+        recipient,
+        amount,
+        start_time,
+        end_time,
+    } = params;
+    let owner = deps.api.addr_validate(&owner)?;
+    let recipient = deps.api.addr_validate(&recipient)?;
+
+    if config.owner == recipient {
```

Comment:
Fair to keep this check, but what is the real purpose of `config.owner`?
To prevent sending to that address or to ensure only that address can create streams on the contract?

If it is the second, then you need

```rust
if config.owner != owner {
  return Err(ContractError::Unauthorised{});
}
```

I don't see any other use of this field and still don't understand why it exists if it is not for this missing check

Reply:
Yeah you are right. Removed this check.